server/controllers/articlesController.js

```diff
@@ -0,0 +1,343 @@
+import db from '../models';
+import generateUniqueSlug from '../helpers/generateUniqueSlug';
+
+const { Profile, Article, Category, User } = db;
+/**
+* @description class will implement CRUD functionalities for articles
+* @class ArticleController
+*/
+class ArticlesController {
+  /**
+   * @param {object} req - Request sent to the route
+   * @param {object} res - Response sent from the controller
+   * @param {object} next - Error handler
+   * @param {object} article
+   * @returns {object} - object representing response message
+   */
+  static async createArticle(req, res, next) {
+
+    const { id } = req.user;
+    const { categoryId } = req.body;
+    const articleSlug = generateUniqueSlug(req.body.title);
+    try {
+
+      const category = await Category.findOne({
+        where: { id: categoryId }
+      });
+
+      const profileId = await Profile.findOne({
+        where: { userId: id }
+      });
+
+      if (!profileId || profileId === null) {
+        return res.status(404).json({
+          success: false,
+          message: 'User does not exist'
+        });
+      }
+      if (!category) {
+        return res.status(400).send({
+          success: false,
+          message: 'Category does not exist',
+        });
+      }
+      const article = await Article.create({
+        title: req.body.title.trim(),
+        body: req.body.body.trim(),
+        slug: articleSlug,
+        published: req.body.published,
+        userId: id,
+        profileId: profileId.dataValues.id,
+        categoryId: req.body.categoryId.trim()
+      });
+
+      return res.status(201).send({
+        success: true,
+        message: 'Article created successfully',
+        article
+      });
+    } catch (error) {
+      return next(error);
+    }
+  }
+
+  /**
+   * @param {object} req - Request sent to the route
+   * @param {object} res - Response sent from the controller
+   *  @param {object} next - Error handler
+   * @param {object} id
+   * @returns {object} - object representing response message
+   */
+  static async getOneArticle(req, res, next) {
+    const { id } = req.params;
+    try {
+      const article = await Article.findOne({
+        where: { id },
+        include: [{
+          model: User,
+          as: 'author',
+          attributes: {
+            exclude: [
```

Comment:
Since we are getting only the username, I believe this will be better to use in this situation 

```
User.findOne({
  attributes: ['username']
});
```